test/bpf/elf-demo.c

```diff
@@ -5,6 +5,7 @@
 #include <bpf/api.h>
 
 #include "lib/utils.h"
+#include <node_config.h>
```

Comment:
It needs to be placed before common.h, since common includes overloadable -> overloadable_skb which references macros defined in node_config.h. Somewhat related to https://github.com/cilium/cilium/pull/27520/commits/7add8a773922823193bb2af4fa65aa7b9434a6d8. I can add a comment here explaining this

Reply:
Not sure if there's any reason against it, but could we put a `#pragma once` in node_config.h and make headers depend directly on the stuff they need instead? Having to worry about include order isn't productive.

Reply:
Seems like this is causing some issues in tests, since there are conditional defines in node_config.h. I think there's going to need to be some order of includes one way or another, unless we refactor some parts of node_config.h. I'll try to find some time to look over it more if this will be a blocker

Reply:
Fair enough, it's a super minor thing.